File: 1. Two Sum/twosum.py
class Solution(object):
  def twoSum(self, nums, target):
    res = list()
    for i in range(len(nums)):
      for j in range(i+1, len(nums)):
        if (nums[i] + nums[j] == target):
          res.append(i)
          res.append(j)
    return res
  def bestTwoSum(self, nums, target):
    res = list()
    d = {}
    for i in range(len(nums)):
      if target - nums[i] in d:
        res.append(d[target- nums[i]])
        res.append(i)
      else:
        d[nums[i]] = i
    return res





p = Solution()
print(p.bestTwoSum([2,7,11,15], 9)) #sample input case 1, you can also use brute force function for same answer
print(p.bestTwoSum([3,2,4], 6)) #sample input case 2

File: 1. Two Sum/twosum.java
import java.util.HashMap;

class Solution {
  public int[] twoSum(int[] nums, int target) {
      for(int i = 0; i < nums.length; i++)
      {
        for(int j = i + 1; j < nums.length; j++)
        {
          if(nums[i] + nums[j] == target) {
            System.out.println(i);
            System.out.println(j);
            return new int[]{i,j};
          }
        }
      }
      return new int[]{-1,-1};
  }

  public int[] bestTwoSum(int[] nums, int target)
  {
    HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
    for(int i = 0; i < nums.length; i++)
    {
      if(hm.containsKey(target- nums[i]))
      {
        System.out.println(i);
        System.out.println(hm.get(target - nums[i]));
        return new int[]{hm.get(target-nums[i]), i};
      }
      else
      {
        hm.put(nums[i], i);
      }
    }
    return new int[]{-1,-1};
  }

  public static void main(String[] args)
  {
    Solution s = new Solution();
    s.bestTwoSum(new int[]{2,7,11,19}, 9);
  }
}